src/Symfony/Component/Mailer/Bridge/Mailgun/RemoteEvent/MailgunPayloadConverter.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Mailgun\RemoteEvent;
+
+use Symfony\Component\RemoteEvent\Event;
+use Symfony\Component\RemoteEvent\Event\Mailer\AbstractMailerEvent;
+use Symfony\Component\RemoteEvent\Event\Mailer\MailerDeliveryEvent;
+use Symfony\Component\RemoteEvent\Event\Mailer\MailerEngagementEvent;
+use Symfony\Component\RemoteEvent\Exception\ParseException;
+use Symfony\Component\RemoteEvent\PayloadConverterInterface;
+
+final class MailgunPayloadConverter implements PayloadConverterInterface
+{
+    /** @return AbstractMailerEvent */
+    public function convert(array $payload): Event
+    {
+        if (in_array($payload['event'], ['failed', 'accepted', 'rejected', 'delivered'])) {
+            if ('failed' === $payload['event']) {
+                $event = 'permanent' === $payload['severity'] ? MailerDeliveryEvent::BOUNCE : MailerDeliveryEvent::DEFERRED;
+            } else {
+                $event = match ($payload['event']) {
+                    'accepted' => MailerDeliveryEvent::RECEIVED,
+                    'rejected' => MailerDeliveryEvent::DROPPED,
+                    'delivered' => MailerDeliveryEvent::DELIVERED,
+                    'blocked' => MailerDeliveryEvent::DROPPED,
+                    default => throw new ParseException(sprintf('Not supported event "%s".', $payload['event'])),
+                };
+            }
+            $wh = new MailerDeliveryEvent($event, $payload['id'], $payload);
```

Comment:
What does `$wh` variable name mean? Could we name it `$event`?

Reply:
I means webhook

Reply:
🤦 ok course it makes sense that it means "webhook" thanks! ... but the code is like this:

```php
$wh = new MailerDeliveryEvent(...);
// ...
$wh = new MailerEngagementEvent(...);
```

Also, we do `return $wh;` in a method with the signature `public function convert(array $payload): Event` so maybe it makes sense to call this `$event`

Reply:
> 🤦 ok course it makes sense that it means "webhook" thanks! ... but the code is like this:

when I first saw the code, I was confused too 😄